packages/dart/npt_flutter/lib/features/profile/bloc/profile_bloc.dart

```diff
@@ -52,6 +52,12 @@ class ProfileBloc extends LoggingBloc<ProfileEvent, ProfileState> {
       profile = null;
     }
 
+    if (event.copyFrom != null) {
+      var json = event.copyFrom!.toJson();
+      json["uuid"] = uuid;
+      profile = Profile.fromJson(json);
+    }
+
```

Comment:
When we call `load or create event` check if there is a profile to copy from.

Reply:
I'm just using the existing json functions to ensure a deep copy by value. Then update the id to the new one, the profile should already have a uuid associated with it.